src/Microsoft.AspNetCore.Blazor/Rendering/ComponentState.cs

```diff
@@ -95,20 +95,6 @@ public void DisposeInBatch(RenderBatchBuilder batchBuilder)
             }
         }
 
-        public void DispatchEvent(EventHandlerInvoker binding, UIEventArgs eventArgs)
-        {
-            if (_component is IHandleEvent handleEventComponent)
-            {
-                handleEventComponent.HandleEvent(binding, eventArgs);
-            }
-            else
-            {
-                throw new InvalidOperationException(
```

Comment:
I'm guessing that this validation is lost now that we've generalized this. I don't have a big concern about it because I haven't seen anyone complain about this error message.

Reply:
This error is no longer applicable as a concept.